### PR TITLE
Migrate to Official Undocked Build System

### PR DESCRIPTION
**Pull request by nibanks**
Moving to the (currently being standardized) official undocked build system to produce signed binaries for all supported architectures (x64, amd64).

- Moves things back to VS project/solution files.
- Fixes a number of warnings when with updated (required) warning level.
- Updates GitHub actions.

### CODE REVIEW

File: src/ntttcp.rc [hunk 1]
```diff
@@ -0,0 +1,12 @@
+//
+//    Copyright (C) Microsoft.  All rights reserved.
+//
+
+#include <windows.h>
+
+#define VER_FILEDESCRIPTION_STR     "NT TCP Performance Tool"
```
Comment (nibanks) [inline src/ntttcp.rc:7]: What would you like to put here?

Comment (anrossi) [reply src/ntttcp.rc:7]: looks good to me.

File: .github/SUPPORT.md [hunk 2]
```diff
@@ -1,25 +1,25 @@
...
-feature request as a new Issue.
-
-For help and questions about using this project, please **REPO MAINTAINER: INSERT INSTRUCTIONS HERE 
-FOR HOW TO ENGAGE REPO OWNERS OR COMMUNITY FOR HELP. COULD BE A STACK OVERFLOW TAG OR OTHER
-CHANNEL. WHERE WILL YOU HELP PEOPLE?**.
-
-## Microsoft Support Policy  
-
-Support for this **PROJECT or PRODUCT** is limited to the resources listed above.
+# TODO: The maintainer of this repo has not yet edited this file
```
Comment (maolson-msft) [inline .github/SUPPORT.md:1]: what changed?

File: .gitattributes.txt [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# Standardize line endings for all text files.
```
Comment (maolson-msft) [inline .gitattributes.txt:1]: Is this file related to the PR?

File: README.md [hunk 4]
```diff
@@ -8,23 +8,14 @@ See [Releases](https://github.com/microsoft/ntttcp/releases) section for binarie
...
-
-```
-mkdir build
-cd build
-cmake ..
-cmake --build . --config Release
-```
-
-The binary will be at `build/Release/ntttcp.exe`
+1. Checkout the repository with submodules.
```
Comment (maolson-msft) [inline README.md:11]: Clone, not checkout

File: .github/SUPPORT.md (hunk 2 above)
Comment (nibanks) [reply .github/SUPPORT.md:1]: Line endings were standardized by the .gitattributes file I believe.

File: src/ntttcp.c [hunk 5]
```diff
@@ -2,7 +2,10 @@
 Copyright (c) Microsoft Corporation.
 */
 
-#define NTTTCP_VERSION "5.39"
+#define NTTTCP_VERSION "5.40" // TODO: Replace with VER_MAJOR "." VER_MINOR
```
Comment (maolson-msft) [inline src/ntttcp.c:5]: can we go ahead and do this? As is, now there are two points of maintenance- here and the new json file

File: src/ntttcp.c [hunk 6]
```diff
@@ -311,9 +317,39 @@ FARPROC lpNtQuerySystemInformation;
 FARPROC lpQueryIdleProcessorCycleTime;
 FARPROC lpGetTcp6Table;
 FARPROC lpSetPerTcp6ConnectionEStats;
-FARPROC lpGetPerTcp6ConnectionEStats;
+typedef
```
Comment (maolson-msft) [inline src/ntttcp.c:320]: Whatever this is, why was it needed for "Get" but not needed for "Set"?

File: src/ntttcp.rc (hunk 1 above)
Comment (maolson-msft) [reply src/ntttcp.rc:7]: lgtm too.

File: src/ntttcp.c (hunk 5 above)
Comment (nibanks) [reply src/ntttcp.c:5]: This requires changes to the undocked repo. I will do it soon, but don't want to block here.

**Pull request merged**